modules/electromagnetics/src/kernels/CoupledFuncDiffusion.C

```diff
@@ -0,0 +1,47 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CoupledFuncDiffusion.h"
+#include "Function.h"
+
+registerMooseObject("ElectromagneticsApp", CoupledFuncDiffusion);
+
+InputParameters
+CoupledFuncDiffusion::validParams()
+{
+  InputParameters params = Kernel::validParams();
+  params.addClassDescription(
+      "Represents a coupled Laplacian term with sign and function coefficients.");
+  params.addParam<FunctionName>("func", 1.0, "Function multiplier for diffusion term.");
+  params.addParam<Real>("sign", 1.0, "Sign of Kernel, if it needs to be changed.");
```

Comment:
could be an enum since it s more a coef than a sign at this point

Reply:
yup - there's a couple instance of this that could be an enum instead. Been pondering the change.